### PR TITLE
Oracle IPC port, by some unsatisfied mapper.

### PR DESCRIPTION
**Pull request by SomeAngryMiner**
## About The Pull Request

As you may see on the title, this is finishes properly (or so i think) the IPC race on the works that got added a while ago, clearing some bugs, adding some features, QOL and such. Bugs expected too!
For now, this is what we get:

-IPC are no longer flashproof. Take that, powergamers.
-IPC are now properly robotic, no longer heal metal by slapping a bandaid on it.
-EMPS fuck you harder, by much, im talking outside and inner damage.
-Organs are now properly robotic, brain goes on chest, as it should do. (this makes headless ipc feasible)
-Liquid solder added, basically synth mannitol.
-System cleaner added, synth charcoal.
-adds vars for races without mouths, such as abductors and the IPC.
-IPC have an innate 5 brute resist, but if that threshol is passed, 1.5 more damage taken, they can still readibily attach limbs again though.
-You can research and make more IPC components on the exofab, therefore letting you properly repair them
-you no longer need boards to reboot ipc, theyll instantly reboot once they have more than 0 hp.
-Robot emotes. mechanically useless, good for getting someone's attention or annoying the captain.
-IPC no longer are stomachless, they now have a powercord to suck on apcs to fill an inner powercell. Make engineers mad as you suck dry the last power of the SM-blown station.
-Deathsound added for IPC, so you can enjoy hearing them die.
-You cant put mechanical brains into organic bodies, and viceversa.
-For now, you cant shove posibrains into ipc chassis. *for now*
-Expect rebalance changes if they are too weak/strong.
-They are properly ignored by medibots. Dont rescue these bastards
-Selective reagent processing. You cant kill them with toxin, but acid still works.
-A lot of spaghetticode, sure to make ike and qwerty cry. stay mad.

## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
imageadd: added some icons and images
imagedel: deleted some icons and images
code: changed some code
refactor: refactored some code
config: changed some config setting
/:cl:

### CODE REVIEW

**Comment by SomeAngryMiner** [comment]
alright, travis shows no errors, time to sit tight and wait for review.

**Comment by ike709** [comment]
Gave this a quick half-asleep skim, seems mostly okay. There's room for improvement but I'm too tired to care at the moment.

My main concern is potential issues with Toxic's terrible IPC code. Could always do a YOLO testmerge and see what happens.

**Comment by Toxici11i** [comment]
Do not merge this until the following issues are fixed. Firstly, the robotics limb trait gives them full augmentations - which makes them space proof, shock resistant and have brute resistances - while keeping the default icons.

Secondly - if you WERE to continue with the robotic trait, remove the current EMP code - It would amplify the effect massively and instantly kill the IPC upon emp (each part takes damage w/ emp)

Also thirdly, the robotics system removes the need for IPC healing code.


Fourthly none of these points actually matter because you just completely copy and pasted oracle code.

**Comment by Toxici11i** [comment]
Other than that, good PR. (also I was joking about the DNM stuff, this is just a better way of doing my current code.)

**Comment by qwertyquerty** [comment]
Anything is better than your way

**Comment by qwertyquerty** [comment]
@SomeAngryMiner this is very important. What method did you use to port it.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1086]: It's important to make sure you have covered any reagents we ourselves have added. Have you done this?

Comment (SomeAngryMiner) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1086]: i checked the entire toxins/medicine/other reagent code, i added flags for the ones i think should affect them of these new ones, i havent checked the alcohol dm though.

Comment (SomeAngryMiner) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1086]: if anything, i was thinking of adding that they can process neurine, or a synth counterpart, because otherwise they are stuck on hallucinations forever if the SM blows up.

**Comment by SomeAngryMiner** [comment]
@qwertyquerty i manually added the missing code how it was on the oracle code, then modified it to new tg standards. Most of the bulk of the old code wasnt needed, because they added either things we already have (robotic surgery, for example), or things that needed an entire /tg/code rewrite, such as most of the posibrain code.

**Review by qwertyquerty — APPROVED** [review]

**Comment by miniusAreas** [comment]
So, because I was from oracle, I'm going to put my two cents in on this because I have seen the state Oracle's IPC's were left in, and saw Sunsetstation try a direct port that's resulted in several of the issues seen in the Discord.

>Can IPC's become lings still (Was fixed, but was also major issue)
>Do limbs retain augments upon de-limbing and re-attachment
>Does System Cleaner / Liquid Solder work (In past, was broken for the longest time)
>Was this a direct port or were the proper refactors made (Oracle was old TG code, and code may need additional refactors to bring it up to speed).
>Can IPC's use Bruise kits/ointment to heal, or Patches (Was issue on Sunset, albeit it broke them completely and hasn't been reported here except for the healing)

Obligitory ping for @qwertyquerty and @someangryminer. This shit was a prevalent issue. If you want further clarification, ping me on the discord and I can hook you up with Ziiro, the person who did Oracle's species coding.

**Comment by SomeAngryMiner** [comment]
-As per se, the ling checks are in other pr someone made before this one, its waiting to be merged.
- limbs do retain augments when deattached and attached, as far as i tested.
- I tested solder and cleaner, and they effectively work, although we have no way other than surgery to cure brain traumas for now, or an equivalent for psicodine in case of hallucinations.
- i refactored the code as much as i can for it to be compatible, although im no experienced coder to fully support that.
- they cant heal using bruise kits or patches, no, since their limbs are robotic.

**Comment by despong** [comment]
add *honk and *sad, important ipc emotes from oracle, the station lost in time

**Pull request merged**
